### PR TITLE
feat: remove Python3.7 support

### PR DESCRIPTION
**Pull request by texastony**
*Issue #, if available:*

*Description of changes:*
- Stop testing or advertising support for py3.7
- Tell PyPi we support 3.10, 3.11, 3.12

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Comment by texastony** [comment]
On Approval, we should remove Python3.7 from required Checks.

File: README.rst [hunk 1]
```diff
@@ -34,7 +34,7 @@ Getting Started
 Required Prerequisites
 ======================
 
-* Python 3.7+
+* Python 3.8+
```
Comment (lucasmcdonald3) [inline README.rst:37]: Could you also do this for `test_vector_handlers/README`?

**Review by lucasmcdonald3 — COMMENTED** [review]
Looks good, there's some other spots in the tests we should clean up as well, but those aren't as important.
Question: Can we also remove `appveyor.yml` as part of this change?

**Comment by texastony** [comment]
> Looks good, there's some other spots in the tests we should clean up as well, but those aren't as important. Question: Can we also remove `appveyor.yml` as part of this change?

All these asks are reasonable.
I am not going to work on them right now, as I need to focus on my sprint work.
But the On Call or anyone can push to this branch to address these.

**Comment by ShubhamChaturvedi7** [comment]
> Looks good, there's some other spots in the tests we should clean up as well, but those aren't as important. Question: Can we also remove `appveyor.yml` as part of this change?

Already removed here: https://github.com/aws/aws-encryption-sdk-python/commit/e28d146e23d261674fe604c4bb88922e970d0ec7

**Review by lucasmcdonald3 — APPROVED** [review]

**Pull request merged**